pages/developers/gha_pure.md

```diff
@@ -128,31 +128,68 @@ later in the upload action for the release job, as well).
 
 And then, you need a release job:
 
+<div class="skhep-bar d-flex m-2" style="justify-content:center;">
+  <button class="skhep-bar-item btn m-2 btn-purple" onclick="openTab('oidc')" id='oidc-btn'>OpenID Connect</button>
+  <button class="skhep-bar-item btn m-2" onclick="openTab('token')" id='token-btn'>Token</button>
+</div>
+
+<div class="skhep-tab" markdown="1" id="oidc">
+
 {% raw %}
 
 ```yaml
-publish:
+upload_all:
   needs: [dist]
+  permissions:
+    id-token: write
   runs-on: ubuntu-latest
```

Comment:
It is also recommended to use GH environments as an extra security measure:
```suggestion
  runs-on: ubuntu-latest
  environment: pypi
```
(if an environment doesn't exist, GH will pre-create it)